components/controller.ts

```diff
@@ -380,6 +380,9 @@ export class Controller {
     public locationsWithETA = new Set<string>()
     public parentsWithETA = new Set<string>()
 
+    // A place for plugins to share data.
+    public blackboard: unknown = {}
```

Comment:
Record<string, unknown> so users don't have to cast the parent, only the member being requested